src/ert/scheduler/lsf_driver.py

```diff
@@ -82,18 +97,26 @@ class RunningJob(BaseModel):
 JOB_ALREADY_FINISHED_BKILL_MSG = "Job has already finished"
 
 
-class _Stat(BaseModel):
-    jobs: Mapping[str, AnyJob]
+def _create_job_class(job_state: JobState) -> AnyJob:
+    return _JOBSTATE_MAP[job_state](job_state)
+
 
+def _parse_jobs_dict(jobs: Mapping[str, str]) -> dict[str, AnyJob]:
+    parsed_jobs_dict: dict[str, AnyJob] = {}
+    for job_id, job_state in jobs.items():
+        parsed_jobs_dict[job_id] = _create_job_class(cast(JobState, job_state))
+    return parsed_jobs_dict
 
-class JobData(BaseModel):
+
+@dataclass
+class JobData:
     iens: int
     job_state: AnyJob
     submitted_timestamp: float
 
 
-def parse_bjobs(bjobs_output: str) -> Dict[str, Dict[str, Dict[str, str]]]:
-    data: Dict[str, Dict[str, str]] = {}
+def parse_bjobs(bjobs_output: str) -> Dict[str, str]:
```

Comment:
Is it possible to tighten typing here from  `str` to something like `JobState`?

Reply:
Yes, but I would need multiple casts in the other parts of the code.